pkg/internal/tracking/context.go

```diff
@@ -28,38 +30,55 @@ const (
 )
 
 type Metadata struct {
-	Version   string    `json:"version,omitempty"`
-	Resource  string    `json:"resource,omitempty"`
-	Operation Operation `json:"operation,omitempty"`
+	SchemaVersion string    `json:"json_schema_version,omitempty"`
+	Version       string    `json:"version,omitempty"`
+	Resource      string    `json:"resource,omitempty"`
+	Datasource    string    `json:"datasource,omitempty"`
+	Operation     Operation `json:"operation,omitempty"`
 }
 
 func (m Metadata) validate() error {
 	errs := make([]error, 0)
+	if m.SchemaVersion == "" {
+		errs = append(errs, errors.New("schema version for metadata should not be empty"))
+	}
 	if m.Version == "" {
-		errs = append(errs, errors.New("version for metadata should not be empty"))
+		errs = append(errs, errors.New("provider version for metadata should not be empty"))
 	}
-	if m.Resource == "" {
-		errs = append(errs, errors.New("resource name for metadata should not be empty"))
+	if m.Resource == "" && m.Datasource == "" {
+		errs = append(errs, errors.New("either resource or data source name for metadata should be specified"))
 	}
 	if m.Operation == "" {
 		errs = append(errs, errors.New("operation for metadata should not be empty"))
 	}
 	return errors.Join(errs...)
 }
 
-func NewMetadata(version string, resource resources.Resource, operation Operation) Metadata {
+// NewTestMetadata is a helper constructor that is used only for testing purposes
+func NewTestMetadata(version string, resource resources.Resource, operation Operation) Metadata {
```

Comment:
Nit: can we move it to a test package?

Reply:
I made it private and any usages outside of the packages were changed to plain struct creation (it was used in one integration test)